src/octoprint/util/yaml_util.py

```diff
@@ -0,0 +1,76 @@
+# -*- coding: utf-8 -*-
+import io
+
+
+def safe_load(file=None, path=None):
+    """
+    Safely and performantly loads yaml data from the given source. Either a
+    path or a file must be passed in.
+
+    :rtype: Union[dict[Hashable, Any], list, None]
+    :type path: str | None
+    :type file: typing.TextIO | None
+    """
+    if path is not None:
+        assert file is None
+        with io.open(path, "rt", encoding="utf-8") as f:
+            return safe_load(file=f)
+
+    if file is not None:
+        assert path is None
+
+    assert (path is not None) or (file is not None)
+
+    import yaml
+
+    try:
+        from yaml import CSafeLoader as SafeLoader
+    except ImportError:
+        from yaml import SafeLoader
+
+    return yaml.load(file, Loader=SafeLoader)
+
+
+def safe_dump(data, file=None, path=None, **kwargs):
+    """
+    Safely and performantly dumps the `data` to yaml. If both file and path
+    are None, the value is returned as a string.
+
+    :rtype: str | None
+    :type data: object
+    :type path: str | None
+    :type file: typing.TextIO | None
+    """
+    if path is not None:
+        assert file is None
+        with io.open(path, "wt", encoding="utf-8") as f:
+            return safe_dump(data, file=f)
+
+    if file is not None:
+        assert path is None
+
+    import yaml
+
+    try:
+        from yaml import CSafeDumper as SafeDumper
+    except ImportError:
+        from yaml import SafeDumper
+
+    return yaml.dump(
+        data,
+        stream=file,
+        Dumper=SafeDumper,
+        allow_unicode=True,  # no good reason not to allow it these days
+        **kwargs
+    )
+
+
+def safe_dump_pretty(data, file=None, path=None):
```

Comment:
Turn into a `pretty` parameter instead and simply adjust `kwargs` passed to `yaml.dump` accordingly.